x-pack/plugin/shutdown/src/internalClusterTest/java/org/elasticsearch/xpack/shutdown/NodeShutdownDelayedAllocationIT.java

```diff
@@ -0,0 +1,241 @@
+        assertBusy(
+            () -> { assertThat(client().admin().cluster().prepareHealth().get().getDelayedUnassignedShards(), equalTo(1)); },
+            2,
+            TimeUnit.SECONDS
+        );
+
+        // And the index should turn green again well within the 30-second timeout
+        ensureGreen(TimeValue.timeValueSeconds(30), "test");
+    }
+
+    public void testShardAllocationTimeoutCanBeChanged() throws Exception {
+        internalCluster().startNodes(3);
+        prepareCreate("test").setSettings(
+            Settings.builder()
+                .put(IndexMetadata.SETTING_NUMBER_OF_SHARDS, 1)
+                .put(IndexMetadata.SETTING_NUMBER_OF_REPLICAS, 1)
+                .put(UnassignedInfo.INDEX_DELAYED_NODE_LEFT_TIMEOUT_SETTING.getKey(), 0) // Disable "normal" delayed allocation
+        ).get();
+        ensureGreen("test");
+        indexRandomData();
+
+        final String nodeToRestartId = findIdOfNodeWithShard();
+        final String nodeToRestartName = findNodeNameFromId(nodeToRestartId);
+
+        {
+            // Mark the node for shutdown with a delay that we'll never reach in the test
+            PutShutdownNodeAction.Request putShutdownRequest = new PutShutdownNodeAction.Request(
+                nodeToRestartId,
+                SingleNodeShutdownMetadata.Type.RESTART,
+                this.getTestName(),
+                TimeValue.timeValueHours(3)
+            );
+            AcknowledgedResponse putShutdownResponse = client().execute(PutShutdownNodeAction.INSTANCE, putShutdownRequest).get();
+            assertTrue(putShutdownResponse.isAcknowledged());
+        }
+
+        // Actually stop the node
+        internalCluster().stopRandomNode(InternalTestCluster.nameFilter(nodeToRestartName));
+
+        // Verify that the shard's allocation is delayed - but with a shorter wait than the reallocation timeout
+        assertBusy(
+            () -> { assertThat(client().admin().cluster().prepareHealth().get().getDelayedUnassignedShards(), equalTo(1)); },
+            2,
+            TimeUnit.SECONDS
+        );
+
+        {
+            // Update the timeout on the shutdown request to something shorter, but that still shouldn't be quite up yet
+            PutShutdownNodeAction.Request putShutdownRequest = new PutShutdownNodeAction.Request(
+                nodeToRestartId,
+                SingleNodeShutdownMetadata.Type.RESTART,
+                this.getTestName(),
+                TimeValue.timeValueSeconds(10)
+            );
+            AcknowledgedResponse putShutdownResponse = client().execute(PutShutdownNodeAction.INSTANCE, putShutdownRequest).get();
+            assertTrue(putShutdownResponse.isAcknowledged());
+        }
+
+        // Verify that the shard's allocation is still delayed
+        assertBusy(
+            () -> { assertThat(client().admin().cluster().prepareHealth().get().getDelayedUnassignedShards(), equalTo(1)); },
+            2,
+            TimeUnit.SECONDS
+        );
```

Comment:
I think we should remove this assert, since a long GC or other CI slowness could cause this to be flaky. I think just the `ensureGreen` below is enough, because it was changed from 3 hours to 10 seconds and the change took effect.